pages/_app.js

```diff
@@ -1,10 +1,31 @@
 import GlobalStyle from "../styles";
+import { SWRConfig } from "swr";
+
+const fetcher = async (url) => {
+  const response = await fetch(url);
+
+  if (!response.ok) {
+    const error = new Error("An error occurred while fetching the data.");
+
+    error.info = await response.json();
+    error.status = response.status;
+    throw error;
+  }
+
+  return response.json();
```

Comment:
To improve this code, you could wrap the fetching logic in a `try` `catch` block.